### PR TITLE
Clarify WebGL1 float color buffer extensions behaviors

### PR DESCRIPTION
**Pull request by zhenyao**
and testing against them

### CODE REVIEW

**Comment by zhenyao** [comment]
@kenrussell @jdashg Please review
@RafaelCintron @grorg @kainino0x FYI

File: extensions/WEBGL_color_buffer_float/extension.xml [hunk 1]
```diff
@@ -38,10 +38,21 @@
       </feature>
 
       <feature>
+        <p>The 32-bit floating-point type <code>RGB32F</code> may become available
+        as a color-renderable format. Renderbuffers can be created in this
+        format. These and textures created with <code>format = RGB</code> and
+        <code>type = FLOAT</code> as specified in <a
+        href="http://www.khronos.org/registry/webgl/extensions/OES_texture_float/">OES_texture_float</a>,
+        can be attached to framebuffer object color attachments for rendering
+	if <code>getFrameBufferStatus()</code> retruns <code>FRAMEBUFFER_COMPLETE</code>.
```
Comment (kenrussell) [inline extensions/WEBGL_color_buffer_float/extension.xml]: typos:
  getFrameBufferStatus() -> checkFramebufferStatus()
  retruns -> returns

File: extensions/EXT_color_buffer_half_float/extension.xml [hunk 2]
```diff
@@ -37,7 +37,9 @@
         Renderbuffers can be created in these formats. These and textures
         created with <code>type = HALF_FLOAT_OES</code>, which will have one
         of these internal formats, can be attached to framebuffer object color
-        attachments for rendering.</p>
+        attachments for rendering. <code>RGBA16F</code> textures are guaranteed
+	to be color-renderable. <code>RGB16F</code> textures are color-renderable
+        if <code>getFramebufferStatus()</code> returns <code>FRAMEBUFFER_COMPLETE</code>.
```
Comment (kenrussell) [inline extensions/EXT_color_buffer_half_float/extension.xml]: getFramebufferStatus() -> checkFramebufferStatus()

Comment (zhenyao) [reply extensions/EXT_color_buffer_half_float/extension.xml]: Done

File: extensions/WEBGL_color_buffer_float/extension.xml (hunk 1 above)
Comment (zhenyao) [reply extensions/WEBGL_color_buffer_float/extension.xml]: Done

**Comment by zhenyao** [comment]
@msc- FYI

File: extensions/WEBGL_color_buffer_float/extension.xml [hunk 3]
```diff
@@ -38,10 +38,21 @@
       </feature>
 
       <feature>
+        <p>The 32-bit floating-point type <code>RGB32F</code> may become available
+        as a color-renderable format. Renderbuffers can be created in this
+        format. These and textures created with <code>format = RGB</code> and
+        <code>type = FLOAT</code> as specified in <a
+        href="http://www.khronos.org/registry/webgl/extensions/OES_texture_float/">OES_texture_float</a>,
+        can be attached to framebuffer object color attachments for rendering
+	if <code>checkFrameBufferStatus()</code> returns <code>FRAMEBUFFER_COMPLETE</code>.
```
Comment (kenrussell) [inline extensions/WEBGL_color_buffer_float/extension.xml]: Capitalization should be:
checkFramebufferStatus()

Comment (zhenyao) [reply extensions/WEBGL_color_buffer_float/extension.xml]: Thanks for catching it. Updated.

File: extensions/EXT_color_buffer_half_float/extension.xml [hunk 4]
```diff
@@ -37,14 +37,16 @@
         Renderbuffers can be created in these formats. These and textures
         created with <code>type = HALF_FLOAT_OES</code>, which will have one
         of these internal formats, can be attached to framebuffer object color
-        attachments for rendering.</p>
+        attachments for rendering. <code>RGBA16F</code> textures are guaranteed
+	to be color-renderable. <code>RGB16F</code> textures are color-renderable
```
Comment (jdashg) [inline extensions/EXT_color_buffer_half_float/extension.xml]: I would say:
RGB16F and RGBA16F effective internal formats become color-renderable, but only RGBA16F is guaranteed to return FRAMEBUFFER_COMPLETE from checkFramebufferStatus().

File: extensions/EXT_color_buffer_half_float/extension.xml [hunk 5]
```diff
@@ -37,14 +37,16 @@
...
+        if <code>checkFramebufferStatus()</code> returns <code>FRAMEBUFFER_COMPLETE</code>.
```
Comment (jdashg) [inline extensions/EXT_color_buffer_half_float/extension.xml]: I think the most correct way to do this is to make both RGB16F and RGBA16F formats color-renderable, but only guarantee not to be FRAMEBUFFER_UNSUPPORTED for RGBA16F. If the implementation doesn't support RGB16F, it would generate FRAMEBUFFER_UNSUPPORTED.

I do think we should deprecate RGB16F, since it's not actually renderable in ES3+ even with extensions.

File: extensions/WEBGL_color_buffer_float/extension.xml [hunk 6]
```diff
@@ -65,6 +76,7 @@
 [NoInterfaceObject]
 interface WEBGL_color_buffer_float {
   const GLenum RGBA32F_EXT = 0x8814;
+  const GLenum RGB32F_EXT = 0x8815;
```
Comment (jdashg) [inline extensions/WEBGL_color_buffer_float/extension.xml]: Don't add this, at least not in this bug. Supporting renderbuffers here is scope creep.

File: extensions/WEBGL_color_buffer_float/extension.xml [hunk 7]
```diff
@@ -38,10 +38,21 @@
       </feature>
 
       <feature>
+        <p>The 32-bit floating-point type <code>RGB32F</code> may become available
+        as a color-renderable format. Renderbuffers can be created in this
+        format. These and textures created with <code>format = RGB</code> and
+        <code>type = FLOAT</code> as specified in <a
+        href="http://www.khronos.org/registry/webgl/extensions/OES_texture_float/">OES_texture_float</a>,
+        can be attached to framebuffer object color attachments for rendering
+	if <code>checkFramebufferStatus()</code> returns <code>FRAMEBUFFER_COMPLETE</code>.
```
Comment (jdashg) [inline extensions/WEBGL_color_buffer_float/extension.xml]: They can always be attached to framebuffers, they just can't always be rendered to.
I would say "may also optionally become color-renderable".

File: sdk/tests/conformance/extensions/oes-texture-float.html [hunk 8]
```diff
@@ -114,24 +114,20 @@
...
-                    return;
-                }
-                testPassed(renderableExtName + " supported. Proceeding to its tests.");
+            var supported = gl.getSupportedExtensions().includes(renderableExtName);
+            if (renderable && !supported) {
+                testFailed("RGBA/FLOAT is color renderable but " + renderableExtName + " not exposed");
+            } else if (supported && !renderable) {
+                testFailed(renderableExtName + " is exposed but RGBA/FLOAT is not color renderable");
+            }
+            if (supported) {
```
Comment (jdashg) [inline sdk/tests/conformance/extensions/oes-texture-float.html]: This should be run even if it's not claimed-supported if it is (incorrectly) implicitly renderable.
You introduce a bug here by not calling getExtension(), which means this will rely on implicit activation. IIRC the idea is that we drop recommendation that new implementations require explicit activation here, but I think we stopped short of requiring implicit activation. If we change this we *must* change the specs to reflect this.

Comment (kenrussell) [reply sdk/tests/conformance/extensions/oes-texture-float.html]: I thought the idea was that the current change to the OES_texture_float and OES_texture_half_float extension specs removing the sentence "Although this feature has historically been allowed..." mandated explicit activation of the renderability extensions if they're supported. If the language isn't strong enough then perhaps this sentence:

"Implementations supporting float rendering via this extension will implicitly enable..."

should also change "will" to "must".

Comment (jdashg) [reply sdk/tests/conformance/extensions/oes-texture-float.html]: I propose going straight to:
Upon activation of this extension, implementations supporting WEBGL_color_buffer_float shall implicitly enable it.
NB: This requirement maintains the historical behavior prior to the differentiation of float renderability from float textures, so as to not break existing content.

Comment (kenrussell) [reply sdk/tests/conformance/extensions/oes-texture-float.html]: Sounds good to me.

File: extensions/EXT_color_buffer_half_float/extension.xml (hunk 5 above)
Comment (zhenyao) [reply extensions/EXT_color_buffer_half_float/extension.xml]: Revised. We can discuss RGB16F removal but I will leave that out of this pull request.

File: extensions/WEBGL_color_buffer_float/extension.xml (hunk 7 above)
Comment (zhenyao) [reply extensions/WEBGL_color_buffer_float/extension.xml]: Revised

File: extensions/WEBGL_color_buffer_float/extension.xml (hunk 6 above)
Comment (zhenyao) [reply extensions/WEBGL_color_buffer_float/extension.xml]: Removed

File: sdk/tests/conformance/extensions/oes-texture-float.html (hunk 8 above)
Comment (zhenyao) [reply sdk/tests/conformance/extensions/oes-texture-float.html]: Addressed in the spec and leave the test as is.

**Comment by zhenyao** [comment]
@jdashg Revised. Please take another look.

File: extensions/WEBGL_color_buffer_float/extension.xml [hunk 9]
```diff
@@ -38,13 +38,14 @@
       </feature>
 
       <feature>
-        <p>The 32-bit floating-point type <code>RGB32F</code> may become available
-        as a color-renderable format. Renderbuffers can be created in this
-        format. These and textures created with <code>format = RGB</code> and
-        <code>type = FLOAT</code> as specified in <a
+        <p>The 32-bit floating-point type <code>RGB32F</code> may also optionally
+	become available as a color-renderable format. Renderbuffers can be created
```
Comment (jdashg) [inline extensions/WEBGL_color_buffer_float/extension.xml]: Renderbuffers cannot be created in this format. Just remove that line from here.

Comment (zhenyao) [reply extensions/WEBGL_color_buffer_float/extension.xml]: Removed

**Comment by zhenyao** [comment]
Merging myself since both @kenrussell  and @jdashg are both OK'ed.
Others, if you have more comments, feel free to let me know and I'll address in a followup.

File: extensions/EXT_color_buffer_half_float/extension.xml [hunk 10]
```diff
@@ -37,14 +37,17 @@
         Renderbuffers can be created in these formats. These and textures
         created with <code>type = HALF_FLOAT_OES</code>, which will have one
         of these internal formats, can be attached to framebuffer object color
-        attachments for rendering.</p>
+        attachments for rendering. Implementations supporting this extension is
+	required to support rendering to <code>RGBA16F</code> format.
+	Applications must check framebuffer completeness to determine if
+	<code>RGA16F</code> is supported.
```
Comment (dmikis) [inline extensions/EXT_color_buffer_half_float/extension.xml]: RGB?
